### PR TITLE
Template markup improvement

### PR DESCRIPTION
**Pull request by Ge11ert**
Hello! I am glad to see you use BEM for class naming convention (at least, it's pertty similar to bem).

But some parts of html template markup contain few mistakes, if we are talking about BEM.

The first (pretty common, but pretty rude) mistake - is having elements of elements. Or having block's grandchilds, if you like.

E.g. in master branch you have a block called `smartbanner`, which has an element `smartbanner__info`, which in his turn also has an element `smartbanner__info__author`.
[Quick link to FAQ with this question](https://en.bem.info/methodology/faq/#why-not-create-elements-of-elements-block__elem1__elem2).

The second what I noticed - markup has a plain `div` tag without any class. Some will judge you for that, some will not.
But AFAIK, you should avoid tags without a class, every should have one. Because if you dont have a class, there's no information about which block this element belongs to.

I am absolutely not shure about `smartbanner__info-container`, I'm not so inventive, so I would like to accept any suggestion, if you have it.

P.S. If these changes will be approved, I think the can go with v-2.0.0, when it comes to light. Because it might be breaking change for those, who rely on classnames.

### CODE REVIEW

**Comment by ain** [comment]
Many thanks for the criticism, it definitely deserves an improvement and it's great you've sent the PR!

You're right about this being a breaking change so it should definitely target [v2.0.0](https://github.com/ain/smartbanner.js/milestone/5).

File: src/smartbanner.js [hunk 1]
```diff
@@ -125,13 +125,13 @@ export default class SmartBanner {
       <a href="javascript:void();" class="smartbanner__exit js_smartbanner__exit" aria-label="${this.closeLabel}"></a>
       <div class="smartbanner__icon" style="background-image: url(${this.icon});"></div>
       <div class="smartbanner__info">
-        <div>
-          <div class="smartbanner__info__title">${this.options.title}</div>
-          <div class="smartbanner__info__author">${this.options.author}</div>
-          <div class="smartbanner__info__price">${this.options.price}${this.priceSuffix}</div>
+        <div class="smartbanner__info-container">
```
Comment (ain) [inline src/smartbanner.js]: I'd suggest a more laconic _copy_ here as in reference to matter that needs to be printed:
```suggestion
        <div class="smartbanner__copy">
```

File: src/smartbanner.js [hunk 2]
```diff
@@ -125,13 +125,13 @@ export default class SmartBanner {
...
+          <div class="smartbanner__title">${this.options.title}</div>
+          <div class="smartbanner__author">${this.options.author}</div>
+          <div class="smartbanner__price">${this.options.price}${this.priceSuffix}</div>
         </div>
       </div>
-      <a href="${this.buttonUrl}" target="_blank" class="smartbanner__button" rel="noopener" aria-label="${this.options.button}"><span class="smartbanner__button__label">${this.options.button}</span></a>
+      <a href="${this.buttonUrl}" target="_blank" class="smartbanner__button" rel="noopener" aria-label="${this.options.button}"><span class="smartbanner__button-text">${this.options.button}</span></a>
```
Comment (ain) [inline src/smartbanner.js]: In the legacy context, let's keep it closer to what it was?
```suggestion
      <a href="${this.buttonUrl}" target="_blank" class="smartbanner__button" rel="noopener" aria-label="${this.options.button}"><span class="smartbanner__button-label">${this.options.button}</span></a>
```

File: src/smartbanner.scss [hunk 3]
```diff
@@ -112,10 +111,10 @@ $black: #000;
     font-size: 18px;
     text-align: center;
     text-decoration: none;
+  }
 
-    &__label {
-      text-align: center;
-    }
+  &__button-text {
```
Comment (ain) [inline src/smartbanner.scss]:
```suggestion
  &__button-label {
```

File: src/smartbanner.scss [hunk 4]
```diff
@@ -191,24 +190,24 @@ $black: #000;
       &:hover {
         background: none;
       }
+    }
 
-      &__label {
-        display: block;
-        padding: 0 10px;
+    .smartbanner__button-text {
```
Comment (ain) [inline src/smartbanner.scss]:
```suggestion
    .smartbanner__button-label {
```

File: test/spec/smartbanner_spec.js [hunk 5]
```diff
@@ -114,52 +114,52 @@ describe('SmartBanner', function() {
       <a href="javascript:void();" class="smartbanner__exit js_smartbanner__exit" aria-label="Close iOS Smart App Banner"></a>
       <div class="smartbanner__icon" style="background-image: url(icon--apple.jpg);"></div>
       <div class="smartbanner__info">
-        <div>
-          <div class="smartbanner__info__title">Smart Application</div>
-          <div class="smartbanner__info__author">SmartBanner Contributors</div>
-          <div class="smartbanner__info__price">FREE - On the App Store</div>
+        <div class="smartbanner__info-container">
```
Comment (ain) [inline test/spec/smartbanner_spec.js]:
```suggestion
        <div class="smartbanner__copy">
```

File: test/spec/smartbanner_spec.js [hunk 6]
```diff
@@ -114,52 +114,52 @@ describe('SmartBanner', function() {
...
+          <div class="smartbanner__title">Smart Application</div>
+          <div class="smartbanner__author">SmartBanner Contributors</div>
+          <div class="smartbanner__price">FREE - On the App Store</div>
         </div>
       </div>
-      <a href="https://itunes.apple.com/us/genre/ios/id36?mt=8" target="_blank" class="smartbanner__button" rel="noopener" aria-label="View"><span class="smartbanner__button__label">View</span></a>
+      <a href="https://itunes.apple.com/us/genre/ios/id36?mt=8" target="_blank" class="smartbanner__button" rel="noopener" aria-label="View"><span class="smartbanner__button-text">View</span></a>
```
Comment (ain) [inline test/spec/smartbanner_spec.js]:
```suggestion
      <a href="https://itunes.apple.com/us/genre/ios/id36?mt=8" target="_blank" class="smartbanner__button" rel="noopener" aria-label="View"><span class="smartbanner__button-label">View</span></a>
```

File: test/spec/smartbanner_spec.js [hunk 7]
```diff
@@ -114,52 +114,52 @@ describe('SmartBanner', function() {
...
 
   const ANDROID_BODY = `<div class="smartbanner smartbanner--android js_smartbanner">
       <a href="javascript:void();" class="smartbanner__exit js_smartbanner__exit" aria-label="Close Android Smart App Banner"></a>
       <div class="smartbanner__icon" style="background-image: url(icon--google.jpg);"></div>
       <div class="smartbanner__info">
-        <div>
-          <div class="smartbanner__info__title">Smart Application</div>
-          <div class="smartbanner__info__author">SmartBanner Contributors</div>
-          <div class="smartbanner__info__price">FREE - In Google Play</div>
+        <div class="smartbanner__info-container">
```
Comment (ain) [inline test/spec/smartbanner_spec.js]:
```suggestion
        <div class="smartbanner__copy">
```

File: test/spec/smartbanner_spec.js [hunk 8]
```diff
@@ -114,52 +114,52 @@ describe('SmartBanner', function() {
...
+          <div class="smartbanner__title">Smart Application</div>
+          <div class="smartbanner__author">SmartBanner Contributors</div>
+          <div class="smartbanner__price">FREE - In Google Play</div>
         </div>
       </div>
-      <a href="https://play.google.com/store" target="_blank" class="smartbanner__button" rel="noopener" aria-label="View"><span class="smartbanner__button__label">View</span></a>
+      <a href="https://play.google.com/store" target="_blank" class="smartbanner__button" rel="noopener" aria-label="View"><span class="smartbanner__button-text">View</span></a>
```
Comment (ain) [inline test/spec/smartbanner_spec.js]:
```suggestion
      <a href="https://play.google.com/store" target="_blank" class="smartbanner__button" rel="noopener" aria-label="View"><span class="smartbanner__button-label">View</span></a>
```

File: test/spec/smartbanner_spec.js [hunk 9]
```diff
@@ -114,52 +114,52 @@ describe('SmartBanner', function() {
...
 
   const UNDEFINED_BODY = `<div class="smartbanner smartbanner--undefined js_smartbanner">
       <a href="javascript:void();" class="smartbanner__exit js_smartbanner__exit" aria-label="Close Smart App Banner"></a>
       <div class="smartbanner__icon" style="background-image: url(icon--apple.jpg);"></div>
       <div class="smartbanner__info">
-        <div>
-          <div class="smartbanner__info__title">Smart Application</div>
-          <div class="smartbanner__info__author">SmartBanner Contributors</div>
-          <div class="smartbanner__info__price">FREE</div>
+        <div class="smartbanner__info-container">
```
Comment (ain) [inline test/spec/smartbanner_spec.js]:
```suggestion
        <div class="smartbanner__copy">
```

File: test/spec/smartbanner_spec.js [hunk 10]
```diff
@@ -114,52 +114,52 @@ describe('SmartBanner', function() {
...
+          <div class="smartbanner__title">Smart Application</div>
+          <div class="smartbanner__author">SmartBanner Contributors</div>
+          <div class="smartbanner__price">FREE</div>
         </div>
       </div>
-      <a href="#" target="_blank" class="smartbanner__button" rel="noopener" aria-label="View"><span class="smartbanner__button__label">View</span></a>
+      <a href="#" target="_blank" class="smartbanner__button" rel="noopener" aria-label="View"><span class="smartbanner__button-text">View</span></a>
```
Comment (ain) [inline test/spec/smartbanner_spec.js]:
```suggestion
      <a href="#" target="_blank" class="smartbanner__button" rel="noopener" aria-label="View"><span class="smartbanner__button-label">View</span></a>
```

File: test/spec/smartbanner_spec.js [hunk 11]
```diff
@@ -114,52 +114,52 @@ describe('SmartBanner', function() {
...
 
   const ANDROID_CUSTOM_DESIGN_BODY = `<div class="smartbanner smartbanner--custom-design js_smartbanner">
       <a href="javascript:void();" class="smartbanner__exit js_smartbanner__exit" aria-label="Close banner"></a>
       <div class="smartbanner__icon" style="background-image: url(icon--google.jpg);"></div>
       <div class="smartbanner__info">
-        <div>
-          <div class="smartbanner__info__title">Smart Application</div>
-          <div class="smartbanner__info__author">SmartBanner Contributors</div>
-          <div class="smartbanner__info__price">FREE - In Google Play</div>
+        <div class="smartbanner__info-container">
```
Comment (ain) [inline test/spec/smartbanner_spec.js]:
```suggestion
        <div class="smartbanner__copy">
```

File: test/spec/smartbanner_spec.js [hunk 12]
```diff
@@ -114,52 +114,52 @@ describe('SmartBanner', function() {
...
+          <div class="smartbanner__title">Smart Application</div>
+          <div class="smartbanner__author">SmartBanner Contributors</div>
+          <div class="smartbanner__price">FREE - In Google Play</div>
         </div>
       </div>
-      <a href="https://play.google.com/store" target="_blank" class="smartbanner__button" rel="noopener" aria-label="View"><span class="smartbanner__button__label">View</span></a>
+      <a href="https://play.google.com/store" target="_blank" class="smartbanner__button" rel="noopener" aria-label="View"><span class="smartbanner__button-text">View</span></a>
```
Comment (ain) [inline test/spec/smartbanner_spec.js]:
```suggestion
      <a href="https://play.google.com/store" target="_blank" class="smartbanner__button" rel="noopener" aria-label="View"><span class="smartbanner__button-label">View</span></a>
```

**Comment by Ge11ert** [comment]
@ain I dont know how to manage with these "suggested changes", actually :) (It seems, GH offers me to create new commits with applied changes, but I need fixup/amend them)
So I just pushed requested fixes into my commits and also performed rebase to your current master in order to resolve conflicts.

**Comment by ain** [comment]
Many thanks! I'll release the current master ast `1.15.0` and then we can jump-start the work for `2.0.0`

**Pull request merged**